model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/InfinispanAuthenticationSessionProvider.java

```diff
@@ -162,36 +170,27 @@ public RootAuthenticationSessionModel getRootAuthenticationSession(RealmModel re
 
     @Override
     public void removeRootAuthenticationSession(RealmModel realm, RootAuthenticationSessionModel authenticationSession) {
-        tx.remove(cache, authenticationSession.getId());
+        SessionUpdateTask<RootAuthenticationSessionEntity> removeTask = Tasks.removeSync();
+        sessionTx.addTask(authenticationSession.getId(), removeTask);
     }
 
     @Override
     public void close() {
 
     }
 
-    public Cache<String, RootAuthenticationSessionEntity> getCache() {
+    public Cache<String, SessionEntityWrapper<RootAuthenticationSessionEntity>> getCache() {
         return cache;
     }
 
-    private record RootAuthenticationSessionUpdater(RootAuthenticationSessionEntity entity,
-                                                    InfinispanAuthenticationSessionProvider provider,
-                                                    RealmModel realm) implements SessionEntityUpdater<RootAuthenticationSessionEntity> {
-
-        @Override
-        public RootAuthenticationSessionEntity getEntity() {
-            return entity;
-        }
-
-        @Override
-        public void onEntityUpdated() {
-            int expirationSeconds = entity.getTimestamp() - Time.currentTime() + SessionExpiration.getAuthSessionLifespan(realm);
-            provider.tx.replace(provider.cache, entity.getId(), entity, expirationSeconds, TimeUnit.SECONDS);
-        }
+    public InfinispanChangelogBasedTransaction<String, RootAuthenticationSessionEntity> getRootAuthSessionTransaction() {
+        return sessionTx;
+    }
 
-        @Override
-        public void onEntityRemoved() {
-            provider.tx.remove(provider.cache, entity.getId());
-        }
+    @Override
+    public void migrate(String modelVersion) {
+        InfinispanConnectionProvider infinispanConnectionProvider = session.getProvider(InfinispanConnectionProvider.class);
```

Comment:
check the `modelVersion` the cache is clear in every version change. 
```java
if ("26.1.0".equals(modelVersion))
```

Reply:
Done!